src/app/tracing/io/data-importer/xlsx-import/shared.ts

```diff
@@ -0,0 +1,616 @@
+        const value = importValue(
+            fromRow,
+            table,
+            columnMapping.fromIndex,
+            columnMapping.type,
+            addIssueCb,
+        );
+        if (value !== undefined) {
+            toRow[columnMapping.toPropId] = value;
+        }
+    }
+}
+
+export function enrichImportIssue(
+    issue: ImportIssue,
+    row: Row,
+    table: Table,
+    invalidateRow: boolean,
+    ref?: string | undefined,
+): ImportIssue {
+    issue = { ...issue };
+    issue.row ??= row?.rowIndex;
+    if (ref !== undefined) {
+        issue.ref ??= ref;
+    }
+    if (invalidateRow) {
+        issue.invalidatesRow = true;
+    }
+    if (issue.col !== undefined && issue.colRef !== undefined) {
+        // col is supposed to be the zero based relative index in the table
+        issue.colRef = table.header.columnHeaders[issue.col];
+        issue.col += table.offset.col;
+        // col is now an absolute 1 based index
+    }
+    return issue;
+}
+
+function getMergedType(
+    types: Set<BasicTypeString>,
+): BasicTypeString | undefined {
+    switch (types.size) {
+        case 0:
+            return undefined;
+        case 1:
+            return Array.from(types)[0];
+        default:
+            return "string";
+    }
+}
+
+export function getOtherColumns(
+    table: Table,
+    startIndex: number,
+    ignoreIndices: number[],
+): ColumnMapping[] {
+    const ignoreIndicesSet = new Set(ignoreIndices);
+    const columnMappings = table.header.columnHeaders.map((h, index) => ({
+        fromIndex: index,
+        toPropId: h.join("_"),
+        type: getMergedType(table.columns[index].types),
+    }));
+
+    const filteredColumnMappings = columnMappings.filter(
+        (col) =>
+            col.fromIndex >= startIndex &&
+            !ignoreIndicesSet.has(col.fromIndex) &&
+            col.type !== undefined,
+    ) as ColumnMapping[];
+    return filteredColumnMappings;
+}
+
+export function getOptionalColumnMapping<T>(
+    table: Table,
+    mappingDef: MappingDef<T>,
+): ColumnMapping[] {
+    const columnMappings: ColumnMapping[] = [];
+    const optionalColRefs = Object.keys(mappingDef) as (keyof MappingDef<T> &
+        string)[];
+    optionalColRefs.forEach((colRef) => {
+        const expectedHeader = mappingDef[colRef]?.header;
+        const columnIndex = table.header.columnHeaders.findIndex(
+            (observedHeader) => _.isEqual(expectedHeader, observedHeader),
+        );
+        if (columnIndex >= 0) {
+            if (table.columns[columnIndex].types.size > 0) {
+                columnMappings.push({
+                    fromIndex: columnIndex,
+                    toPropId: colRef,
+                    type: mappingDef[colRef]!.type,
+                });
+            }
+        }
+    });
+    return columnMappings;
+}
+
+export function getCleanedStringOrUndefined(
+    value: CellValue | undefined,
+): string | undefined {
+    if (typeof value === "string") {
```

Comment:
You want to return the string, as long as it is not empty. In all other cases, you want to return undefined, correct? I'd go for early returns here, e.g. sth like this

```
if (typeof value !== 'string' || value.trim().length === 0) {
    return undefined;
}

return value.trim();
```

Reply:
Done.